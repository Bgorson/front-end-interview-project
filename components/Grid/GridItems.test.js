import {cleanup, fireEvent, render} from '@testing-library/react';
import renderer from 'react-test-renderer';
import GridItem from './GridItems';

afterEach(cleanup);
const fullData= {id:100, name:'Testing Wines', products:[{
    id:1,
    title: 'foo', 
    assets: [{'url':'https://d3lbhvavid9616.cloudfront.net/assets/products/54767/1599850348070.jpg'}]
},{
    id:2,
    title: 'bar',
    assets: [{'url':'https://d3lbhvavid9616.cloudfront.net/assets/products/54767/1599850348070.jpg'}]}

]};
const missingData= {id:100, name:'No Product Wines', products:[]};

it('View more will disappear when reaching max items', () => {
    const {queryByText} = render(
        <GridItem item= {fullData} amountToView= {1}/>,
    );
    expect(queryByText(/No Items are currently available at this time/i)).toBe(null);
    expect(queryByText(/View More Products/i)).toBeTruthy();

    fireEvent.click(queryByText(/View More Products/i));

    expect(queryByText(/View More Products/i)).toBe(null);
});
it('should render an empty state if no items available', () => {
    const {queryByText} = render(
        <GridItem item= {missingData} amountToView= {1}/>,
    );
    expect(queryByText(/No Items are currently available at this time/i)).toBeTruthy();
    
});
it('should render correctly', () => {
    const component = renderer.create(<GridItem item= {fullData} amountToView= {1}/>).toJSON();

    expect(component).toMatchSnapshot();
});