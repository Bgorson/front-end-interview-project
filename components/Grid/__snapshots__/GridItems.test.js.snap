// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
Array [
  <div
    className="GridItems__StyledGridItem-sc-17lm9e-0 gNSRve"
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "inline-block",
          "margin": 0,
          "maxWidth": "100%",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "maxWidth": "100%",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          role="presentation"
          src="data:image/svg+xml;charset=utf-8,<svg width=\\"500\\" height=\\"500\\" xmlns=\\"http://www.w3.org/2000/svg\\" version=\\"1.1\\"/>"
          style={
            Object {
              "display": "block",
              "maxWidth": "100%",
            }
          }
        />
      </div>
      <img
        alt="foo"
        className="itemPicture"
        decoding="async"
        src="/_next/image?url=https%3A%2F%2Fd3lbhvavid9616.cloudfront.net%2Fassets%2Fproducts%2F54767%2F1599850348070.jpg&w=1200&q=75"
        srcSet="/_next/image?url=https%3A%2F%2Fd3lbhvavid9616.cloudfront.net%2Fassets%2Fproducts%2F54767%2F1599850348070.jpg&w=768&q=75 1x, /_next/image?url=https%3A%2F%2Fd3lbhvavid9616.cloudfront.net%2Fassets%2Fproducts%2F54767%2F1599850348070.jpg&w=1024&q=75 2x, /_next/image?url=https%3A%2F%2Fd3lbhvavid9616.cloudfront.net%2Fassets%2Fproducts%2F54767%2F1599850348070.jpg&w=1200&q=75 3x"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "visibility": "visible",
            "width": 0,
          }
        }
      />
    </div>
    <div
      className="GridItems__StyledItemText-sc-17lm9e-1 iHzoOt"
    >
      <span>
        foo
      </span>
      <p
        className="GridItems__StyledText-sc-17lm9e-2 lppsyc"
      >
        $
        .00
      </p>
    </div>
  </div>,
  <p
    className="GridItems__StyledTextButton-sc-17lm9e-3 jIheFY"
    onClick={[Function]}
  >
    View More Products
  </p>,
]
`;
